### ISSUE TITLE
intermittent CredentialError 

### ISSUE THREAD

**Issue by ismaan1998** [Azure/azure-iot-sdk-python#792]
**Details**
Team, We've deployed a solution where we're receiving the telemetry after each 10 seconds from our edge device. 
The connection from Azure Resources to edge device is site to site.
We've deployed a custom module to retrieve the telemetry from a device.
Language used - Python 3.8 
Operating System (Edge device): Ubuntu 18.04
edgeHub version - 1.2
edgeAgent version - 1.2
Authentication Type - Symmetric Keys

**Issue**
After deploying our custom module to edge device, We're able to receive the telemetry from edge device to IoT Hub. It is working perfectly as expected, but after a couple of days it suddenly stopped sending the telemetry for 13 hours and started again automatically. 

We've another custom module which send the logs data to Azure Log Analytics, and when we took a look on the logs in between those 13 hours, we got to know that we're receiving an exception while trying to send the telemetry using **IoTHubModuleClient's** **send_message**() function. 
![image](https://user-images.githubusercontent.com/30122168/122719698-33213200-d28c-11eb-9ce4-97963f9056c4.png)
CredentialError('Credentials invalid, could not connect') caused by UnauthorizedError('Connection Refused: not authorised.')

**Question** - If the problem is with the credentials, how we've received telemetry before and after that 13 hours period? 

**Solution we've tried** -
As per the documentation https://docs.microsoft.com/en-us/azure/iot-edge/troubleshoot-common-errors?view=iotedge-2020-11. 
We've set the OptimizeForPerformance as False, When we were not receiving the logs. But it wasn't the issue as we got the exception mentioned above.

Here are couple of screenshots for more information - 
![image](https://user-images.githubusercontent.com/30122168/122721501-5e0c8580-d28e-11eb-9650-b2603cb83e74.png)

![image](https://user-images.githubusercontent.com/30122168/122721991-dd9a5480-d28e-11eb-80ee-09279419b97c.png)

Let us know if anything needed to debug this issue from our side.

**Comment by cartertinney** [comment]
I've traced this bug, and what's happening under the surface is that you are successfully sending a connection request, but the CONNACK that comes back from Hub is refusing your request. I'm going to reach out to the Hub and Edge teams to see if they have some insight into why that might be.

In the meantime, can you tell me if you are using any optional configurations for the client (optional keyword arguments at instantiation via `.create_from_edge_enviornment()`)? Or are you simply using the default settings?

**Comment by ismaan1998** [comment]
@cartertinney I'm using the default settings.

**Comment by cartertinney** [comment]
@ismaan1998 

Has this happened more than once, or just that one time? Furthermore, can you tell me the date that this occurred and the deployment region of your IoTHub?

**Comment by ismaan1998** [comment]
@cartertinney 
It was on 21st of June (Just Once). Region is East US

**Comment by cartertinney** [comment]
@ismaan1998 

Do you have the exact time it went down, and when it resumed? Any logs you could send? Also what is the name of your device, module, and Hub? We're trying to trace what happened here, so any extra information you can provide would be helpful.

**Comment by ismaan1998** [comment]
Went down - June 20, 12:39 PM 
Resumed - June 21, 01.39 AM
IoT Hub name - iot-ncf-002
Device name - business-edge

Logs contain some sensitive information @cartertinney, So can't share it here. In logs, everything is as expected, the only thing that was extra, I've shared it above (CredentialError).

**Comment by ismaan1998** [comment]
@cartertinney  Any update ?

**Comment by cartertinney** [comment]
@ismaan1998 Not yet I'm afraid, I'm reaching out to see if anything was going on at that day and time that could explain this issue, but have not yet heard back.

I assume this still hasn't reoccurred yet?

**Comment by ismaan1998** [comment]
@cartertinney No, It hasn't reoccurred yet. However we've implemented the custom solution to store the dropped telemetry. But just wanted to make sure what was the actual issue so we can be ready for the next time.

**Comment by cartertinney** [comment]
@ismaan1998

Is it possible that your device IP address was blocked or not whitelisted? Someone looked into this issue, and the error causing your downtime was a result of your source IP being blocked/

**Comment by ismaan1998** [comment]
I've checked, No we haven't changed anything at that specific time. The connection between the edge device and IoT hub is over site to site (and using private endpoints), Don't think we need to specify the IPs that can connect to IoT Hub. But as we're  not able to replicate the issue from our side (as well as we have implemented the backup messaging solution to store the telemetry in the backend when there is connection drop). We're good to close this task and will can be re-opened if we found this issue again.

Thanks, @cartertinney  for looking into this and helping us out.
